warehouse/forklift/legacy.py

```diff
@@ -1354,6 +1275,38 @@ def file_upload(request):
                 },
             )
 
+        # If the user provided attestations, store them
+        if attestations:
+            attestations_db_models = []
+            for attestation in attestations:
+                with tempfile.NamedTemporaryFile() as tmp_file:
+                    tmp_file.write(attestation.model_dump_json().encode("utf-8"))
+
+                    attestation_digest = hashlib.file_digest(
+                        tmp_file, "sha256"
+                    ).hexdigest()
+                    database_attestation = DatabaseAttestation(
+                        file=file_, attestation_file_sha256_digest=attestation_digest
+                    )
+
+                    storage.store(
+                        database_attestation.attestation_path,
+                        tmp_file.name,
+                        meta=None,
+                    )
+
+                    attestations_db_models.append(database_attestation)
```

Comment:
Thought: something to consider is defining an appropriate `AttestationService` or similar to abstract these steps into a single API. The advantage to this would be easier unit testing (since we could write a development variant of that service or mock it) + making it harder to miss one of these stateful calls.

In other words, this would all become roughly:

```py
db_attestations.append(attestation_service.persist_attestation(attestation))
```

...and that would do the model creation + file services + etc. under the hood 🙂

Reply:
I'm closing this thread in favor of Dustin's comment below.